src/elm/Model.elm

```diff
@@ -39,3 +40,43 @@ initial =
         , Resource.build "Bicycle" 47 1.07 (Currency 12000)
         ]
     }
+
+
+purchaseResource : Int -> Resource.Resource -> Model -> Model
+purchaseResource count resource model =
+    let
+        transaction =
+            Resource.purchase count resource
+
+        newResource =
+            Resource.applyTransaction transaction
+
+        newResources =
+            replaceResource resource newResource model.resources
+
+        finalCost =
+            Resource.transactionCost transaction
+    in
+    if model |> canPayFor finalCost then
```

Comment:
I know we'd discussed having a "0 transaction" that would not add the resource and have a cost of 0 currency. That eliminates the conditional here but does make it harder for us to have special logic (other than no-op) for the the case where the user can't afford an item.